docs: provide example for enabling mesh on a per namespace basis (#12255)
hashicorp/consul: website/content/docs/k8s/installation/install.mdx

You can learn what settings are available by running `helm inspect values hashicorp/consul`
or by reading the [Helm Chart Reference](/docs/k8s/helm).

> #### Minimal `values.yaml` for Consul Service Mesh
> 
> The minimal settings to enable [Consul Service Mesh]((/docs/k8s/connect)) would be captured in the following `vaules.yaml` config file:

Reviewer: ```suggestion
#### Minimal `config.yaml` for Consul Service Mesh

The minimal settings to enable [Consul Service Mesh]((/docs/k8s/connect)) would be captured in the following `config.yaml` config file:
```
The code blocks below use `config.yaml` as the filename.

...
```

#### Enable Consul Service Mesh on select namespaces

> By default, Consul Service Mesh is enabled on all namespaces within a Kubernetes cluster. You can restrict this to a subset of namespaces by specifying a `namespaceSelector` that matches a label attached to each namespace denoting whether to enable Consul service mesh.

Reviewer 2: We exclude a couple by default: https://github.com/hashicorp/consul-k8s/blob/5027066696e2a95fea6342da96081c1581574149/charts/consul/values.yaml#L1797
```suggestion
By default, Consul Service Mesh is enabled on almost all namespaces within a Kubernetes cluster. You can restrict this to a subset of namespaces by specifying a `namespaceSelector` that matches a label attached to each namespace denoting whether to enable Consul service mesh.
```

<CodeBlockConfig filename="config.yaml">

```yaml
global:
  name: consul
connectInject:
  enabled: true

>   default: true

Reviewer 2: Should we call this out a bit more that setting `default` to `true` is required for this to work?

Author: Thanks added!

...
```

#### Enable Consul Service Mesh on select namespaces

> By default, Consul Service Mesh is enabled on all namespaces within Kubernetes. However if you'd like to control which namespaces allow connect-inject to occur, you can alternatively use a label on a namespace to denote where to enable Consul Service Mesh. 
> 

Reviewer: ```suggestion
By default, Consul Service Mesh is enabled on all namespaces within a Kubernetes cluster. You can restrict this to a subset of namespaces by specifying a `namespaceSelector` that matches a label attached to each namespace denoting whether to enable Consul service mesh.
```
